nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -506,13 +506,17 @@ bool RegulatedPurePursuitController::isCollisionImminentExtendedSearch()
     global_plan_.poses.begin(),
     last_pose_it,
     std::back_inserter(extended_collision_check_path.poses),
-    [](geometry_msgs::msg::PoseStamped & poseStamped) {
+    [this](geometry_msgs::msg::PoseStamped & poseStamped) {
       // Modify Z For visualization in rviz so it is drawn between global plan and collision arc
-      poseStamped.pose.position.z = 0.005;
-      return poseStamped;
+      geometry_msgs::msg::PoseStamped transformed_pose;
+      poseStamped.header = global_plan_.header;
+      transformPose(costmap_ros_->getGlobalFrameID(), poseStamped, transformed_pose);
+      transformed_pose.pose.position.z = 0.005;
+      return transformed_pose;
     }
   );
-  extended_collision_check_path.header = global_plan_.header;
+  extended_collision_check_path.header.frame_id = costmap_ros_->getGlobalFrameID();
```

Comment:
Do not repeat: assign `costmap_ros_->getGlobalFrameID()` to a variable.

Reply:
The first assign is for the frame_id for the poses in the path. And he second one is the frame_id of the path itself.